servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ProxyTunnel.java

```diff
@@ -105,72 +106,76 @@ int connectCount() {
-                    out.flush();
-
-                    final InputStream cin = clientSocket.getInputStream();
-                    Future<Void> f = executor.submit(() -> {
-                        copyStream(out, cin);
-                        return null;
-                    });
-                    copyStream(clientSocket.getOutputStream(), in);
-                    f.get(); // wait for the copy of proxy client input to server output to finish copying.
-                }
-            } else {
-                throw new RuntimeException("Unrecognized initial line: " + initialLine);
+    private void handleRequest(final Socket serverSocket, final String initialLine) throws IOException {
+        if (initialLine.startsWith(CONNECT_PREFIX)) {
+            final int end = initialLine.indexOf(' ', CONNECT_PREFIX.length());
+            final String authority = initialLine.substring(CONNECT_PREFIX.length(), end);
+            final String protocol = initialLine.substring(end + 1);
+            final int colon = authority.indexOf(':');
+            final String host = authority.substring(0, colon);
+            final int port = Integer.parseInt(authority.substring(colon + 1));
+
+            try (Socket clientSocket = new Socket(host, port)) {
+                connectCount.incrementAndGet();
+                final OutputStream out = serverSocket.getOutputStream();
+                out.write((protocol + " 200 Connection established\r\n\r\n").getBytes(UTF_8));
+                out.flush();
+
+                executor.submit(() -> {
+                    try {
+                        copyStream(out, clientSocket.getInputStream());
+                    } catch (IOException e) {
+                        LOGGER.debug("Error copying clientSocket input to serverSocket output", e);
+                    } finally {
+                        try {
+                            // We are simulating a proxy that doesn't do half closure. The proxy should close the server
+                            // socket as soon as the server read is done. ServiceTalk's CloseHandler is expected to
+                            // handle this gracefully (and delay FIN/RST until requests/responses complete).
+                            serverSocket.close();
+                        } catch (IOException e) {
+                            LOGGER.debug("Error closing serverSocket", e);
+                        }
+                    }
+                });
+                copyStream(clientSocket.getOutputStream(), serverSocket.getInputStream());
+
+                // Don't wait on the clientSocket input to serverSocket output copy to complete. We want to simulate a
+                // proxy that doesn't do half closure and that means we should close as soon as possible. ServiceTalk's
+                // CloseHandler should handle this gracefully (and delay FIN/RST until requests/responses complete).
```

Comment:
Consider referencing test classes that depend on this expectation: GracefulConnectionClosureHandlingTest and ConnectionCloseHeaderHandlingTest